enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,314 @@
+       client_cert /var/run/ocp-collector/secrets/elasticsearch/cert
+       ca_file /var/run/ocp-collector/secrets/elasticsearch/cacert
+       <buffer>
+		@type file
+		path '/var/lib/fluentd/elasticsearch'
+       </buffer>
+     </store>
+  </match>
+  <match **>
+     @type copy
+     <store>
+       @type elasticsearch
+       @id elasticsearch
+       client_key /var/run/ocp-collector/secrets/elasticsearch/key
+       client_cert /var/run/ocp-collector/secrets/elasticsearch/cert
+       ca_file /var/run/ocp-collector/secrets/elasticsearch/cacert
+       retry_tag "retry_elasticsearch"
+     </store>
+ </match>
+</label>
+<label @SECUREFORWARD_OFFCLUSTER>
+   ...
+</label>
+```
+
+Following is a diagram which identifies the workflow of a message through fluentd as captured by the above configuration.
+
+![https://docs.google.com/drawings/d/18FD4c1pBz4e8eJIH37yUwM40ew69g8h1cqPEfOLsuyY/edit?usp=sharing](images/log-forwarding-message-flow.png)
+
+### Cluster Logging Operator
+* The operator generates output configuration for the collector
+* The operator specifies in the collector daemonset the secrets needed to communicate to each endpoint:
+
+   `/var/run/ocp-collector/secrets/<secretName>/{cert,key,cacert}`
+
+    where:
+
+    * `secretName` is the name of a secret in the `openshift-logging` namespace.
+    * `cert`, `key`, `cacert` are files in the secret
+#### Collector
+* Collectors will be modified to be remove endpoint config specific logic from the start script ; configuration is assumed to be correct and used as provided by the cluster-logging-operator
+* Extract all configuration from after the source config point to final destination into the collector configuration.
+
+### Test Plan
+#### Unit testing
+* Log forwarding will add unit tests to provide adequate coverage for all changes
+* BDD unit testing will be added to unit testing to make tests goals more explicit,  readable, and obvious
+#### Integration and E2E tests
+* Test no regressions after upgrade when log forwarding exists
+* Tests to verify there are no regressions in functionality between pre and post logforwarding
+* Tests to verify log forwarding is writing logs to an Elasticsearch instance not managed by cluster logging
+* Tests to verify log forwarding is writing logs to an Elasticsearch instance managed by cluster logging
+* Tests to verify log forwarding is writing logs to a Fluentd instance via secure forward that is not managed by cluster logging
+
+### Graduation Criteria
+
+##### Tech Preview
+
+* Ability to store logs to the cluster managed log aggregator without defining log forwarding (e.g. status quo)
+* Explicitly define and document log sources and their meanings (e.g. `logs.apps`, `logs.infra`)
+* Explicitly define supported targets (e.g. `elasticsearch`, `fluentd`)
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
```

Comment:
Preferred method for feedback?

Reply:
> Preferred method for feedback?

@sichvoge Is there a specific mechanism to identify in this proposal?

Reply:
I'd suggest to collect feedback on the cluster-logging-operator repo. Just create a new issue and we can start a discussion if necessary. WDYT?